app/models/annotation/AnnotationPrivateLink.scala

```diff
@@ -0,0 +1,84 @@
+package models.annotation
+
+import com.scalableminds.util.accesscontext.DBAccessContext
+import com.scalableminds.util.tools.Fox
+import com.scalableminds.webknossos.schema.Tables.{AnnotationPrivatelinks, _}
+
+import javax.inject.Inject
+import play.api.libs.json.{JsValue, Json, OFormat}
+import play.api.mvc.PlayBodyParsers
+import slick.jdbc.PostgresProfile.api._
+import slick.lifted.Rep
+import utils.{ObjectId, SQLClient, SQLDAO}
+
+import scala.concurrent.ExecutionContext
+
+case class AnnotationPrivateLink(_id: ObjectId,
+                                 _annotation: ObjectId,
+                                 accessToken: String,
+                                 expirationDateTime: Option[Long],
+                                 isDeleted: Boolean = false)
+
+object AnnotationPrivateLink {
+  implicit val jsonFormat: OFormat[AnnotationPrivateLink] = Json.format[AnnotationPrivateLink]
+}
+
+case class AnnotationPrivateLinkParams(_annotation: String, expirationDateTime: Option[Long])
+
+object AnnotationPrivateLinkParams {
+  implicit val jsonFormat: OFormat[AnnotationPrivateLinkParams] = Json.format[AnnotationPrivateLinkParams]
+}
+
+class AnnotationPrivateLinkService @Inject()()(implicit ec: ExecutionContext, val bodyParsers: PlayBodyParsers) {
+
+  val DefaultAnnotationListLimit = 1000
+  def publicWrites(annotationPrivateLink: AnnotationPrivateLink): Fox[JsValue] =
+    Fox.successful(Json.toJson(annotationPrivateLink))
```

Comment:
Then again, I just saw that the objectIds are sent as json objects, which is not very nice. So maybe let’s keep the writes here but instead make it explicit so that it can unpack the objectId and send them as strings only.
I created refactoring issue https://github.com/scalableminds/webknossos/issues/6375 to resolve this properly later